src/pivot-table/hooks/use-items-rendered-handler/utils/get-column-pages.ts

```diff
@@ -13,16 +13,23 @@ const isMissingColumnData = (measureData: MeasureData, x: number, y: number, hei
 const canMergePages = (prevPage: EngineAPI.INxPage, page: EngineAPI.INxPage) =>
   prevPage.qLeft + prevPage.qWidth === page.qLeft && prevPage.qTop === page.qTop && prevPage.qHeight === page.qHeight;
 
-const getColumnPages = (measureData: MeasureData, qLeft: number, qTop: number, qWidth: number, qHeight: number) => {
+const getColumnPages = (
+  pageInfo: PageInfo,
+  measureData: MeasureData,
+  x: number,
+  y: number,
+  width: number,
+  height: number,
+) => {
   const pages = [];
 
-  for (let left = qLeft; left < qLeft + qWidth; left++) {
-    if (isMissingColumnData(measureData, left, qTop, qHeight)) {
+  for (let currX = x; currX < x + width; currX++) {
+    if (isMissingColumnData(measureData, currX, y, height)) {
       const prevPage = pages[pages.length - 1] as EngineAPI.INxPage | undefined;
       const page = {
-        qLeft: left,
-        qTop,
-        qHeight,
+        qLeft: currX,
+        qTop: Math.max(0, y + pageInfo.page * pageInfo.rowsPerPage),
+        qHeight: height,
         qWidth: 1,
       };
```

Comment:
a small thing but worth mentioning:
IMO, previous naming with `qLeft` is more readable since they are taken from same terminology in pivot table fetch page properties. 

The problem with new variable names are that hard to keep track which one is which while passing them and renaming on the function signature

Reply:
same for `get-row-pages` as well

Reply:
I never could come up with a good names here. The problem is that the `y` parameter is not really equivalent to `qTop` because of pagination. I tried naming it `pageY`, but did not like that as it can be confussed with the objects that are of types `EngineAPI.INxPage`.

I'll give it some more thought.

Reply:
Ended up going back to the "qLeft" naming. But for "qTop" I named it "pageTop" to be clear that it does not represent the actual "qTop" value.